### PR TITLE
[R TST] Set up R Travis Environment

### PR DESCRIPTION
**Pull request by terrytangyuan**
Resolved some namespace issues but haven't fixed everything yet. Will start a separate PR later.

### CODE REVIEW

**Comment by tqchen** [comment]
Some comment:
- Please change sudo back to ```false```
   - Let us only do osx build on R
- Consider upgrade the roxygen to 5.0.0, so the roxygen message won't be duplicated and we won't touch so many Rd files.
- Please do a squash of the recent commits so the updates on Rd files don't need to be touch

**Comment by terrytangyuan** [comment]
@tqchen Fixed according to your comments. Sorry about the roxygen thing. I am using my old mac and I forgot to update it.

**Comment by tqchen** [comment]
You want to uncomment the r_task grid to open the build.

**Comment by terrytangyuan** [comment]
Yes but I still need to fix the error during final installation step. I will work on it later along with other more robust test cases in a separate PR.

**Comment by tqchen** [comment]
OK, this is merged. cc @thirdwing

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +17,7 @@ env:
   - TASK=cpp_test
   # run tests/python
   - TASK=python_test
+  #- TASK=r_test
```
Comment (thirdwing) [inline .travis.yml]: We need to uncommet this line to start R test.

Comment (tqchen) [inline .travis.yml]: seems current config is yet to work

Comment (thirdwing) [inline .travis.yml]: https://travis-ci.org/dmlc/mxnet

There seems no R testing task.

Comment (tqchen) [inline .travis.yml]: Yes, according to @terrytangyuan The current config is not yet fully completed and there were some error in final setups, so it is not enabled

Comment (terrytangyuan) [inline .travis.yml]: Oh BTW, I forgot to uncomment the slack notification in Travis file. Did Travis send a lot of messages to your channel before I commented out? (Hope didn't bother you guys a lot...)

**Pull request merged**